src/views/projects/SourceBrowser/Changeset.svelte

```diff
@@ -39,6 +40,27 @@
     }
     return s.join(", ");
   };
+
+  let selectedFilename: string | undefined = undefined;
+  let selectedHunkLine: string | undefined = undefined;
+
+  $: {
+    updateSelectedLine();
+  }
+
+  function updateSelectedLine() {
+    const fragment = window.location.hash.substr(1);
+    if (fragment.match(/.*:H(\d*)L(\d*)L?(\d*)/)) {
+      [selectedFilename, selectedHunkLine] = fragment.split(":");
+    }
+  }
+  onMount(async () => {
+    window.addEventListener("hashchange", updateSelectedLine);
+  });
+
+  onDestroy(() => {
+    window.removeEventListener("hashchange", updateSelectedLine);
+  });
```

Comment:
Is there a reason why we can’t move this code to the `FileDiff` component? For one, we should try to minimize the API surface (i.e. the properties) for components. And we should also try to keep code that implements a single feature close together.

Reply:
it's because of the event listener, here we only add one on `Changeset`, but if we mount them on `FileDiff`, it'll be linear and depend on number of files involved in diff.

Reply:
> it's because of the event listener, here we only add one on `Changeset`, but if we mount them on `FileDiff`, it'll be linear and depend on number of files involved in diff.

I see. But I wouldn’t be to concerned about the performance of adding event listeners. Browsers are pretty good at handling a ton of listeners. For example your register two click listeners for every line your render and this is a lot more than  one for every file.

So splitting the code between the components has almost no effect on performance but does make the code a lot more complex and brittle.

Reply:
updated in https://github.com/radicle-dev/radicle-interface/pull/942/commits/1f80f35a0967e226ffea940f640d819e35da456e